src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -154,7 +155,14 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
         .setEvents(events);
 
       outputStream.reset();
-      addEventsRequest.writeJson(compressor.newStreamCompressor(outputStream));
+
+      // NOTE: We use countingStream since we also need access to the raw serialized payload size before the compression
+      OutputStream compressorStream = compressor.newStreamCompressor(outputStream);
+      CountingOutputStream countingStream = new CountingOutputStream(compressorStream);
```

Comment:
Declare this as:
`CountingOutputStream countingStream = new CountingOutputStream(compressor.newStreamCompressor(outputStream));`

I think this will address the SonarCode issue.

Reply:
Resolved in 1ed121bc24b4a46be3cb72bc6ce0a7eba8b5ea00.